src/components/ChatDetailsQuickActionsBar.tsx

```diff
@@ -14,45 +15,26 @@ type ChatDetailsQuickActionsBarProps = {
 
 function ChatDetailsQuickActionsBar({report}: ChatDetailsQuickActionsBarProps) {
     const styles = useThemeStyles();
-    const [isLastMemberLeavingGroupModalVisible, setIsLastMemberLeavingGroupModalVisible] = useState(false);
     const {translate} = useLocalize();
     const isPinned = !!report.isPinned;
     return (
         <View style={[styles.flexRow, styles.ph5, styles.mb5]}>
             <View style={[styles.flex1, styles.pr3]}>
-                <ConfirmModal
-                    danger
-                    title={translate('groupChat.lastMemberTitle')}
-                    isVisible={isLastMemberLeavingGroupModalVisible}
-                    onConfirm={() => {
-                        setIsLastMemberLeavingGroupModalVisible(false);
-                        Report.leaveGroupChat(report.reportID);
-                    }}
-                    onCancel={() => setIsLastMemberLeavingGroupModalVisible(false)}
-                    prompt={translate('groupChat.lastMemberWarning')}
-                    confirmText={translate('common.leave')}
-                    cancelText={translate('common.cancel')}
-                />
                 <Button
-                    onPress={() => {
-                        if (Object.keys(report?.participants ?? {}).length === 1) {
-                            setIsLastMemberLeavingGroupModalVisible(true);
-                            return;
-                        }
-
-                        Report.leaveGroupChat(report.reportID);
-                    }}
-                    icon={Expensicons.Exit}
+                    onPress={() => Report.togglePinnedState(report.reportID, isPinned)}
+                    icon={Expensicons.Pin}
                     style={styles.flex1}
-                    text={translate('common.leave')}
+                    text={isPinned ? translate('common.unPin') : translate('common.pin')}
                 />
             </View>
             <View style={[styles.flex1]}>
                 <Button
-                    onPress={() => Report.togglePinnedState(report.reportID, isPinned)}
-                    icon={Expensicons.Pin}
+                    onPress={() => {
+                        Navigation.navigate(ROUTES.REPORT_WITH_ID_DETAILS_SHARE_CODE.getRoute(report?.reportID ?? ''));
+                    }}
+                    icon={Expensicons.QrCode}
                     style={styles.flex1}
-                    text={isPinned ? translate('common.unPin') : translate('common.pin')}
+                    text={translate('common.share')}
```

Comment:
well, we need to add back the `if (!isGroupDMChat)` to this part now... I thought that was clear from my last comment sorry if not.

Reply:
Would you mind posting a screen shot of this change? I want to make sure it's what @shawnborton expects. Thanks!

Reply:
I think we're working on a component for this that has all of that baked in. @grgia does that sound right?

Reply:
Check this PR here: https://github.com/Expensify/App/pull/41972

Just want to make sure we aren't duplicating work.

Reply:
@shawnborton thanks for noticing! It looks like this PR will conflict with https://github.com/Expensify/App/pull/41972

cc @grgia

Reply:
Thanks for the heads up @shawnborton 

I think we're good here though. We just need to fix some conflicts now @ZhenjaHorbach. I've instructed @cdOut to pause the next PR they mentioned since we don't really need it.

@grgia might want to do a quick scan to make sure there aren't any other improvements with open issues that snuck into that design doc, but don't think there are a ton. They should all be in the #vip-split project and categorized by Group Chats.